adapters/huaweiads/huaweiads.go

```diff
@@ -548,34 +562,49 @@ func getNativeFormat(adslot30 *adslot30, openRTBImp *openrtb2.Imp) error {
 		if asset.Img != nil {
 			if asset.Img.Type == native1.ImageAssetTypeMain {
 				numMainImage++
-				if asset.Img.H != 0 && asset.Img.W != 0 {
-					width = asset.Img.W
-					height = asset.Img.H
-				} else if asset.Img.WMin != 0 && asset.Img.HMin != 0 {
-					width = asset.Img.WMin
-					height = asset.Img.HMin
+				if numFormat > 1 && asset.Img.H != 0 && asset.Img.W != 0 && asset.Img.WMin != 0 && asset.Img.HMin != 0 {
+					formats = append(formats, format{asset.Img.W, asset.Img.H})
+				}
+				if numFormat == 1 && asset.Img.H != 0 && asset.Img.W != 0 && asset.Img.WMin != 0 && asset.Img.HMin != 0 {
+					result := filterPopularSizes(popularSizes, asset.Img.W, asset.Img.H, "ratio")
+					for i := 0; i < len(result); i++ {
+						formats = append(formats, format{result[i]["w"], result[i]["h"]})
+					}
+				}
```

Comment:
should add json framework test to cover above code path

Reply:
Hi @onkarvhanumante , thanks for all feedbacks and review 🙌 It's updated.

Reply:
thanks @ahmetfaruk59  

could you link jsontest where `numFormat`count is 1 and `asset.Img.W` is 0 and `asset.Img.H` is 0

If jsontest is not available then should add one for above scenario

Reply:
thanks @onkarvhanumante . actually nativeThreeImageIncludeIcon test file covers this scenario but i added a new test file only this case in last commit. thanks again for your feedback 🙌

Reply:
Hi @onkarvhanumante , hope you are ok. I thought this PR would automatically merged for version 2.0.2.  When will it be merged soon? It was a bit urgent. thanks a lot.